### PR TITLE
add www.zooniverse.org nginx server block

### PR DESCRIPTION
**Pull request by camallen**
allow serving requests proxied from the www.zooniverse.org domain without incorrect host name redirection. 
Extract the common nginx config blocks to it's own config map and reuse these across different server name blocks.

# Review checklist

- [ ] First, the most important one: is this PR small enough that you can actually review it? Feel free to just reject a branch if the changes are hard to review due to the length of the diff.
- [ ] If there are any migrations, will they the previous version of the app work correctly after they've been run (e.g. the don't remove columns still known about by ActiveRecord).
- [ ] If anything changed with regards to the public API, are those changes also documented in the `apiary.apib` file?
- [ ] Are all the changes covered by tests? Think about any possible edge cases that might be left untested.

### CODE REVIEW

File: kubernetes/deployment-production.tmpl [hunk 1]
```diff
@@ -1,96 +1,72 @@
...
+---
+apiVersion: v1
+data:
+  nginx.conf: |+
+    upstream docker-panoptes {
+      server localhost:81;
+    }
+    
+    server {
+      server_name www.zooniverse.org;
```
Comment (adammcmaster) [inline kubernetes/deployment-production.tmpl:55]: If these server blocks are all the same you don't really need more than one. You can group multiple names together:

```
      server_name www.zooniverse.org panoptes.zooniverse.org signin.zooniverse.org;
```

File: kubernetes/deployment-production.tmpl [hunk 2]
```diff
@@ -1,96 +1,72 @@
...
+      include /etc/nginx/ssl.default.conf;
+      include /etc/nginx-sites/panoptes-nginx-common.conf;
```
Comment (adammcmaster) [inline kubernetes/deployment-production.tmpl:57]: And if you group them together you probably don't need to split this out into a separate file.

File: kubernetes/deployment-production.tmpl [hunk 3]
```diff
@@ -230,6 +206,8 @@ spec:
               mountPath: "/static-assets"
             - name: panoptes-nginx-config
               mountPath: "/etc/nginx-sites"
+            - name: panoptes-nginx-common.conf
+              mountPath: "/etc/nginx-sites"
```
Comment (adammcmaster) [inline kubernetes/deployment-production.tmpl:210]: If you're going to keep this in a separate file you'll want to mount it to `/etc/nginx/` instead. `/etc/nginx-sites/*` is included by the main nginx config and including this file there will cause an error.

File: kubernetes/deployment-production.tmpl (hunk 1 above)
Comment (camallen) [reply kubernetes/deployment-production.tmpl:55]: my reading suggests that this will use the `www.zooniverser.org` as the $server_name variable which will then be passed to the upstream via host header, https://github.com/zooniverse/panoptes/blob/325d71760298c701efa5db1202e8b5b4232112b1/kubernetes/deployment-production.tmpl#L40

this will break redirects as per the issue outlined in https://github.com/zooniverse/operations/issues/459

If we can verify how the $server_name variable is filled when using one server block (wildcard subdomain / regex are options here) then i'm happy to collapse them - i did look into collapsing them but couldn't ensure the $server_name var will be correctly set.

Comment (adammcmaster) [reply kubernetes/deployment-production.tmpl:55]: In my testing it seems to set `$server_name` to the first hostname value:

```nginx
server {
	server_name test testing;
	rewrite ^.*$ http://$server_name redirect;
}
```

```
$ curl -v -H "Host: test" localhost:8080/ | grep Location
< Location: http://test
$ curl -v -H "Host: testing" localhost:8080/ | grep Location
< Location: http://test
```

So I guess that's not what we want.

We'll need to keep them separated as you have done then in that case. (Another option might be to use `$http_host` but that could be dangerous if we accidentally deploy with no default server block, as that would open it to every host name.)

Comment (camallen) [reply kubernetes/deployment-production.tmpl:55]: thanks for doing the actual testing Adam - appreciated. FWIW the security issue is fixed in rails 6+ so we can remove this nginx header config if / when we get there.

**Comment by camallen** [comment]
> This looks OK now, but I would also recommend doing a similar refactoring to the staging config. I know staging doesn't need the multiple hostnames but it would be good practice to keep the two configs as similar as possible.

Will do before merging - in fact i might get a staging PR out to ensure this works as expected.
